test/support/requests_generator.ex

```diff
@@ -0,0 +1,77 @@
+defmodule MongoosePushWeb.Support.RequestsGenerator do
+  # APIv2/3 requests
+  def mandatory_fields() do
+    StreamData.fixed_map(%{
+      "alert" =>
+        StreamData.fixed_map(%{
+          "title" => nonempty_string(),
+          "body" => nonempty_string()
+        }),
+      "service" => service()
+    })
+  end
+
+  def optional_fields() do
+    StreamData.optional_map(%{
+      "alert" =>
+        StreamData.optional_map(%{
+          "badge" => positive_integer(),
+          "click_action" => nonempty_string(),
+          "tag" => nonempty_string(),
+          "sound" => nonempty_string()
+        }),
+      "data" => data(),
+      "mode" => mode(),
+      "priority" => priority(),
+      "mutable_content" => boolean(),
+      "tags" => tags(),
+      "topic" => nonempty_string(),
+      "time_to_live" => positive_integer()
+    })
+  end
+
+  def device_id() do
+    nonempty_string()
+  end
+
+  def mandatory_field() do
+    one_of_strings([:service, :title, :body])
+  end
+
+  # basic types
+  defp nonempty_string() do
+    StreamData.string(:alphanumeric) |> StreamData.filter(fn x -> String.length(x) > 0 end)
```

Comment:
What about something like (or whatever the correct syntax is)
Suggested change
```suggestion
    StreamData.string(:alphanumeric, min_length: 1)
```
Note that StreamData.string/2 can take the same options than list_of/1

Reply:
good catch 😀